### PR TITLE
`combine()` uses tidy dots. closes #3407.

### LINKED ISSUE #3407: Tidy dots in bind_rows(), bind_cols() and combine()
to resolve ambiguity when these functions are used with just one argument, see https://github.com/tidyverse/dplyr/pull/3387#issuecomment-372076766:

``` r
library(tidyverse)
combine(list(1))
#> [1] 1
combine(list(1), list(2))
#> [[1]]
#> [1] 1
#> 
#> [[2]]
#> [1] 2
```

> Created on 2018-03-12 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

The first call should also return a list, if unpacking is desired, the user can use `combine(!!!list(1))` (which currently doesn't work).

Implement new verbs, or support a compatibility argument via *pkgconfig*.

### PR DESCRIPTION
**Pull request by romainfrancois**
```
> chunks <- list(1,2,3)
> combine(!!!chunks)
[1] 1 2 3
```

### CODE REVIEW

**Comment by krlmlr** [comment]
It looks incomplete, because I'm still seeing (with this PR):

``` r
library(tidyverse)
combine(list(1))
#> [1] 1
combine(list(1), list(2))
#> [[1]]
#> [1] 1
#> 
#> [[2]]
#> [1] 2
```

> Created on 2018-05-29 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

The first call should return a list of one element for consistency. This is a slightly breaking change, but I think it's worth it.

**Comment by romainfrancois** [comment]
I'll revisit later. What I tried quickly broke too many tests...

**Comment by romainfrancois** [comment]
Sure. Go for it.

File: R/bind.r [hunk 1]
```diff
@@ -147,11 +147,12 @@ bind_cols <- function(...) {
 #' If `combine()` it is called with exactly one list argument, the list is
 #' simplified (similarly to `unlist(recursive = FALSE)`. `NULL` arguments are
 #' ignored. If the result is empty, `logical()` is returned.
+#' Use [vctrs::vec_c()] if you never want to unlist.
```
Comment (krlmlr) [inline R/bind.r]: @hadley: Okay to point to _vctrs_ from the dev version of _dplyr_?

**Comment by krlmlr** [comment]
I did a force-push to this branch, sorry for the inconvenience.

File: R/bind.r (hunk 1 above)
Comment (hadley) [reply R/bind.r]: Yes. Maybe we should explicitly mark as questioning?

**Comment by romainfrancois** [comment]
> I did a force-push to this branch, sorry for the inconvenience.

Sure. no problem.

File: R/bind.r (hunk 1 above)
Comment (krlmlr) [reply R/bind.r]: Done.

**Comment by romainfrancois** [comment]
what's the state of this @krlmlr ?

**Comment by krlmlr** [comment]
I think it's good to go, I remember we agreed to keep the behavior of `combine()` and point to `vec_c()` instead.

**Pull request merged**